lcls1_producers/smd_producer.py

```diff
@@ -138,55 +74,58 @@ def define_dets(run):
             
             # Analysis functions
             # ROIs:
-            if detname in ROIs:
-                if not isinstance(ROIs[detname], list):
-                    ROIs[detname] = [ ROIs[detname] ]
-                for ROI in ROIs[detname]:
-                    det.addFunc(ROIFunc(**ROI))
+            if detname in roi_args:
+                for ROI in roi_args[detname]:
+                    pjs = ROI.pop('pj',[])
+                    rfunc = ROIFunc(**ROI)
+                    for pj in pjs:
+                        rfunc.addFunc(projectionFunc(**pj))
+                    det.addFunc(rfunc)
             # Azimuthal binning
-            if detname in az:
-                det.addFunc(azimuthalBinning(**az[detname]))
-            if detname in az_pyfai:
-                det.addFunc(azav_pyfai(**az_pyfai[detname]))
-            # Photon count
-            if detname in phot:
-                det.addFunc(photonFunc(**phot[detname]))
+            if detname in azint_args:
+                for azint in azint_args[detname]:
```

Comment:
Do we have a case where we need to run several azav on the same detector? That's odd, but if we do I guess I am fine with this